src/clib/pioc_support.cpp

```diff
@@ -540,6 +540,70 @@ int end_adios2_step(file_desc_t *file, iosystem_desc_t *ios)
 
     return PIO_NOERR;
 }
+
+adios2_variable* spio_define_adios2_variable(iosystem_desc_t *ios, file_desc_t *file, adios2_io *io,
+                                             const char *name, const adios2_type type, const size_t ndims,
+                                             const size_t *shape, const size_t *start, const size_t *count,
+                                             const adios2_constant_dims constant_dims)
+{
+    /* Define the variable in ADIOS2 */
+    adios2_variable* variable = adios2_define_variable(io, name, type, ndims, shape, start, count, constant_dims);
+
+#ifdef _SPIO_ADIOS_USE_COMPRESSION
+    assert(ios != NULL && file != NULL);
+    /* Skip compression for scalar variables (ndims == 0), see https://github.com/ornladios/ADIOS2/issues/4390 */
+    /* FIXME: Consider adding a new IO type, PIO_IOTYPE_ADIOSC, and use it as a condition for applying compression */
+    if (variable != NULL && ios->compression_operator != NULL && ndims > 0 /* && file->iotype == PIO_IOTYPE_ADIOSC */)
+    {
+        size_t operation_index = 0;
+        adios2_error adiosErr = adios2_error_none;
+
+        #if defined(ADIOS2_HAVE_BLOSC2)
+        /* Compression operations: specify compressor type, compression level, and shuffle option */
+        adiosErr = adios2_add_operation(&operation_index, variable, ios->compression_operator, "compressor", "zstd"); /* Blosc2 compressor type (e.g., zstd) */
+        if (adiosErr != adios2_error_none)
+        {
+            pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
+                    "Failed to add Blosc2 compression (compressor=zstd) to variable %s (adios2_error=%s)",
+                    name, convert_adios2_error_to_string(adiosErr));
+        }
+
+        /* FIXME: The following code supports adding extra compression operations to the variable,
+           which might cause run time errors for BP5 engine if the variable is involved in multiple adios2_put calls.
+           As it is uncertain whether multiple puts will apply in this context, the code is commented out for now.
+           See https://github.com/ornladios/ADIOS2/issues/4390 for further discussion.
+         */
+        #if 0
```

Comment:
Can we comment this block instead of "#if 0" ?

Reply:
Actually we can use adios2_set_operation_parameter() here, as suggested by ADIOS developers. I will rebase this PR.